Logging/FileLogHandler.cs

```diff
@@ -20,14 +20,13 @@
 namespace QuantConnect.Logging
 {
     /// <summary>
-    /// Provides an implementation of <see cref="ILogHandler"/> that writes all log messages to a file on disk.
+    /// Provides an implementation of <see cref="BaseLogHandler"/> that writes all log messages to a file on disk.
     /// </summary>
-    public class FileLogHandler : ILogHandler
+    public class FileLogHandler : BaseLogHandler
     {
         private bool _disposed;
```

Comment:
This may need to be marked as `volatile` now since it's being read from a background thread but its value could be changed from a foreground thread which makes the read in `WriteMessage` potentially stale.

Another option is to change it to a `CancellationTokenSource` which manages the thread memory barriers correctly and efficiently.